pkg/clustermanager/cluster_manager.go

```diff
@@ -139,11 +147,59 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 		Name: managementCluster.Name,
 	}
 
-	if err := c.applyCluster(ctx, managementCluster, workloadCluster, clusterSpec, provider, false, types.WithNodeRef()); err != nil {
+	cpContent, mdContent, err := provider.GenerateClusterApiSpecForCreate(ctx, workloadCluster, clusterSpec)
+	if err != nil {
+		return nil, fmt.Errorf("error generating capi spec: %v", err)
+	}
+
+	content := templater.AppendYamlResources(cpContent, mdContent)
+
+	if err = c.writeCapiSpecFile(clusterSpec.ObjectMeta.Name, content); err != nil {
```

Comment:
curious, what is this for?

Reply:
since we removed the write from provider code, this is the only place to write CAPI spec to local